src/snowflake/cli/_plugins/nativeapp/sf_facade_exceptions.py

```diff
@@ -0,0 +1,62 @@
+from click import ClickException
+from snowflake.cli.api.constants import ObjectType
+from snowflake.connector import DatabaseError, Error
+
+
+def handle_unclassified_error(err: Error | Exception, context: str):
+    """
+    Handles exceptions that are not caught by categorized exceptions in SQLFacade
+    @param err: connector error or base exception
+    @param context: message to add context to exception
+    """
+    message = f"{context} {str(err)}"
+    if isinstance(err, DatabaseError):
+        raise UnknownSQLError(message) from err
+
+    if isinstance(err, Error):
+        raise UnknownConnectorError(message) from err
+
+    # Not a connector error
+    raise Exception(f"Unclassified exception occurred. {message}") from err
```

Comment:
I'm not comfortable exposing this message to users. If these are Python exceptions, we should just let them bubble up. If it's something, it should be wrapped. We should consider just using the original exception here, provided that re-raising it maintains the original stack trace.

Reply:
really good point. I'll just pass the message and re-raise it (the point of the message is to add context from wherever we called this `handle_unclassified_error`.

Reply:
Would it make sense to raise a custom error instead of `Exception` here? I don't think we will have access to the message on the dashboard, so I could see it being more helpful to see something like `UnknownFacadeError` so we know more specifically where a spike of errors could be coming from, instead of just `Exception`

Reply:
Not a custom error, but the original exception. These are basically python exceptions, we'd want to see what they are and where they come from (not in telemetry, but the user should be able to obtain a trace and submit a bug report)